### ISSUE TITLE
Ansible collection ignored when running playbook on AWX 23.0.0 or 23.2.0

### ISSUE THREAD

**Issue by Keith-Nowosielski-SJSU** [ansible/awx#14518]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

Similar or same issue as : https://github.com/ansible/awx/issues/10236

When syncing project role from AWX collections placed in roles/requirements.yml or collections/requirements.yml are not being installed used. 

The sync stdout indicates the collection was detected but must be manually installed (WHY?). 

`        "The requirements file '/var/lib/awx/projects/_31__devel_vca_snapshot/roles/requirements.yml' contains collections which will be ignored. To install these collections run 'ansible-galaxy collection install -r' or to install both at the same time run 'ansible-galaxy install -r' without a custom install path.",`

The only way to even view this with these instructions is to open the output while manually syncing the project. Otherwise it will only say it's ignoring the collections to be installed. 

Logging into the awx-ee pod and running the collection installation command as instructed installs the collection, however the modules are still missing when running the template. This is either because the collection isn't installed in the expected location or because the awx-ee pod isn't the same as the control-plane which I'm not sure what exactly that's supposed to be, the k8s host(s)?? 

### AWX version

23.2.0 

### Select the relevant components

- [X] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [X] Collection
- [X] CLI
- [X] Other

### Installation method

kubernetes

### Modifications

no

### Ansible version

2.15.4

### Operating system

Ubuntu 22.04.3 LTS

### Web browser

Chrome

### Steps to reproduce

1. Configure the Inventory 
2. Configured github project and sync the project. (notice the collection doesn't sync) 
3. Configure the job template 
4. Run the job template  
5. Verify Roles and Collections download are configured in settings (repeat from step 2). 

### Expected results

The collection should automatically download and install to run the job template. 

### Actual results

[job_246.txt](https://github.com/ansible/awx/files/12786056/job_246.txt)
[project_update_249.txt](https://github.com/ansible/awx/files/12786057/project_update_249.txt)

### Additional information

First attempted on AWX 23.0.0 then upgraded to AWX 23.2.0
Attempted to manually install the collection. - didn't work
Attempted workaround on https://github.com/ansible/awx/issues/10236 - didn't work
Attempted workaround on https://github.com/ansible/awx/issues/9768 - didn't work

**Comment by ksquires** [comment]
I'm not experiencing this issue using 23.2.0.   I have roles defined in `roles/requirements.yml` and collections defined in `collections/requirements.yml` and both are getting installed at project sync time for me.

**Comment by Keith-Nowosielski-SJSU** [comment]
> I'm not experiencing this issue using 23.2.0. I have roles defined in `roles/requirements.yml` and collections defined in `collections/requirements.yml` and both are getting installed at project sync time for me.

Yup, I was just about to close this because I tried putting the collections in collections/requirements.yml and it didn't seem to work there either. Troubleshooting further after I opened this bug report I noticed that the new folder and file weren't actually being updated into my repo. I had requirements in my gitignore file for some reason. 

I still think it should work from roles/requirements.yml and I have confirmed that adding a collection to roles/requirements.yml does work with git sourced collections using git+url, it's only galaxy collections that it won't work.
